causal_testing/json_front/json_class.py

```diff
@@ -87,6 +87,8 @@ def setup(self, scenario: Scenario):
                 "attribute with a dataframe before calling .setup()"
             )
         self._populate_metas()
+        self.data_collector = ObservationalDataCollector(
```

Comment:
Hardcoding an `ObservationalDataCollector` seems dangerous. What if the user wants to use an `ExperimentalDataCollector`? We've only ever used one once, but that in the CTF paper to show that you can do it, so that means we need to keep support for it, unfortunately

Reply:
I'm not sure if there is a nice way of making the JSON frontend support both. 

The `ExperimentalDataCollector` has abstract methods that need to be defined somewhere, which goes against the JSON frontend design of being a command line utility. There is the additional issue of where would the parameters required to instantiate an `ExperimentalDataCollector` be specified.

Reply:
If we do keep it as only `ObservationalDataCollector`s though, I think it is worth making it very clear that the frontend only supports that style

Reply:
Could we do the same thing as we did with estimators and that, and just have a dict to convert between strings and classes?

Reply:
We could, but the user would still have to make a separate python script in order to implement the `run_system_with_input_configuration` method from the `ExperimentalDataCollector`. Then would need to manually invoke the Json frontend or copy and paste the argparser stuff across.

It just feels a bit messier, but doable if it is useful.

Reply:
I'll make this an issue for a future PR for now, as this is already a big PR and it is a bit out of the scope. The previous JSON implementation also had this hard-coded so for now it isn't changing the functionality.